internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,348 @@
+				Schema: map[string]*pluginsdk.Schema{
+					"type": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"value": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"display_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"host_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(customlocations.HostTypeKubernetes),
+			}, false),
+		},
+	}
+}
+
+func (r CustomLocationResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r CustomLocationResource) ModelObject() interface{} {
+	return &CustomLocationResourceModel{}
+}
+
+func (r CustomLocationResource) ResourceType() string {
+	return "azurerm_extended_custom_location"
+}
+
+func (r CustomLocationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model CustomLocationResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+
+			id := customlocations.NewCustomLocationID(subscriptionId, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+
+			if model.Authentication != nil {
+				auth := model.Authentication[0]
+				customLocationProps.Authentication = &customlocations.CustomLocationPropertiesAuthentication{
+					Type:  &auth.Type,
+					Value: &auth.Value,
+				}
+			}
+
+			if model.ClusterExtensionIds != nil {
+				customLocationProps.ClusterExtensionIds = &model.ClusterExtensionIds
+			}
+
+			if model.DisplayName != "" {
+				customLocationProps.DisplayName = &model.DisplayName
+			}
+
+			if model.HostResourceId != "" {
+				customLocationProps.HostResourceId = &model.HostResourceId
+			}
+
+			if model.HostType != "" {
+				hostType := customlocations.HostType(model.HostType)
+				customLocationProps.HostType = &hostType
+			}
+
+			if model.Namespace != "" {
+				customLocationProps.Namespace = &model.Namespace
+			}
+
+			props := customlocations.CustomLocation{
+				Id:         utils.String(id.ID()),
+				Location:   model.Location,
+				Name:       utils.String(model.Name),
+				Properties: &customLocationProps,
+			}
```

Comment:
Could we update to use `pointer.To` instead of `utils.String`, please also update other places.

Reply:
Will do.